python_modules/dagster/dagster/_core/definitions/metadata/table.py

```diff
@@ -301,7 +301,64 @@ class TableColumnLineage(
         ],
     )
 ):
-    """Represents the lineage of column outputs to column inputs for a tabular asset."""
+    """Represents the lineage of column outputs to column inputs for a tabular asset.
+
+    Args:
+        deps_by_column (Mapping[str, Sequence[TableColumnDep]]): A mapping from column names to
+            the columns that the column depends on.
+
+    Examples:
+        Defining column lineage at materialization time, where the resulting asset's `new_column_a`
+        is derived from `column_a` in `source_one` and `source_two`, and `new_column_b` is derived
+        from `column_b` in `source_one` and `source_two`:
+
+            .. code-block:: python
+
+                from dagster import (
+                    AssetKey,
+                    MaterializeResult,
+                    TableColumnDep,
+                    TableColumnLineage,
+                    asset,
+                )
+
+
+                @asset(
+                    deps=[
+                        AssetKey("source_one"),
+                        AssetKey("source_two"),
+                    ]
+                )
+                def my_asset():
+                    yield MaterializeResult(
+                        metadata={
+                            "dagster/column_lineage": TableColumnLineage(
+                                deps_by_column={
+                                    "new_column_a": [
+                                        TableColumnDep(
+                                            asset_key=AssetKey("source_one"),
+                                            column_name="column_a",
+                                        ),
+                                        TableColumnDep(
+                                            asset_key=AssetKey("source_two"),
+                                            column_name="column_a",
+                                        ),
+                                    ],
+                                    "new_column_b": [
+                                        TableColumnDep(
+                                            asset_key=AssetKey("source_one"),
+                                            column_name="column_b",
+                                        ),
+                                        TableColumnDep(
```

Comment:
consider leaving out "source_two" here to make it clear that a column doesn't need to depend on every upstream asset